packages/11ty/_plugins/iiif/manifest/index.js

```diff
@@ -0,0 +1,162 @@
+      .map((item) => {
+        let id, motivation
+        switch(true) {
+          case !!item.src:
+            id = path.parse(src).name
+            motivation = 'painting'
+            break
+          case !!item.text:
+            id = item.label.split(' ').join('-').toLowerCase()
+            motivation = 'text'
+            break
+          default:
+            break;
+        }
+        /**
+         * @todo create text and image annotation bodies
+         */
+        this.createAnnotation({
+          id: item.id || id,
+          label: this.getAnnotationLabel(item),
+          motivation
+        })
+      })
+  }
+
+  get choices() {
+    const choices = this.figure.annotations
+      .flatMap(({ items }) => items)
+      .filter((item) => !item.target && item.src)
+
+    const items = choices.map((item) => {
+      const { src } = item
+      if (!src) {
+        error(`Invalid annotation on figure ID "${this.figure.id}". Annotations must have a "src" or "text" property`)
+      }
+      const label = this.getAnnotationLabel(item)
+      const { name } = path.parse(src)
+      const choiceId = new URL([this.iiifConfig.inputDir, src].join('/'), process.env.URL).href
+      const format = mime.lookup(src)
+      const choice = {
+        id: choiceId,
+        format,
+        height: this.canvas.height,
+        type: 'Image',
+        label: { en: [label] },
+        width: this.canvas.width
+      }
+      if (this.figure.preset === 'zoom') {
+        const serviceId = new URL([this.iiifConfig.outputDir, name, this.iiifConfig.imageServiceDirectory].join('/'), process.env.URL)
+        choice.service = [
+          {
+            id: serviceId,
+            type: 'ImageService3',
+            profile: 'level0'
+          }
+        ]
+      }
+      return choice
+    })
+
+    return this.createAnnotation({
+      body: {
+        items,
+        type: 'Choice',
+      },
+      id: 'choices',
+      motivation: 'painting',
+    })
+  }
+
+  async create() {
+    const { height, width } = await this.getCanvasMetadata()
+    const manifest = builder.createManifest(this.manifestId, (manifest) => {
+      manifest.addLabel(this.figure.label, this.iiifConfig.locale)
+      manifest.createCanvas(this.canvas.id, (canvas) => {
+        canvas.height = height
+        canvas.width = width
+        if (this.annotations) {
+          this.annotations.forEach((item) => {
+            canvas.createAnnotation(item.id, item)
+          })
+        }
+        if (this.choices) {
+          canvas.createAnnotation(this.choices.id, this.choices)
+        }
+      })
+    })
+    this.manifest = manifest
+  }
+
+  createAnnotation({ body, id, motivation }) {
+    return {
+      body: body,
+      id: [this.canvas.id, id].join('/'),
+      motivation: motivation,
+      type: 'Annotation'
+    }
+  }
+
+  getAnnotationLabel({ label, src }) {
+    const filename = src ? path.parse(src).name : null
+    return label || titleCase(filename)
+  }
```

Comment:
Can `label` not be a property of object returned by `createAnnotation`?

Reply:
Not exactly since choices and annotations have a slightly different data structure, it made more sense to me to just have it as a helper method